app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java

```diff
@@ -258,6 +261,21 @@ public void deleteUploadSchemaById(@Nonnull StudyIdentifier studyIdentifier, @No
         }
         return uploadSchema;
     }
+    
+    /** {@inheritDoc} */
+    @Override
+    public @Nonnull List<UploadSchema> getUploadSchemaAllRevisions(@Nonnull StudyIdentifier studyIdentifier, @Nonnull String schemaId) {
+        List<DynamoUploadSchema> uploadSchemas = getUploadSchemaAllRevisionsNoThrow(studyIdentifier.getIdentifier(), schemaId);
+        if (uploadSchemas.isEmpty()) {
+            throw new EntityNotFoundException(UploadSchema.class, String.format(
+                "Upload schema not found for study %s, schema ID %s", studyIdentifier.getIdentifier(), schemaId));
+        }
+        List<UploadSchema> schemas = new ArrayList<>();
+        for (DynamoUploadSchema schema : uploadSchemas) {
+            schemas.add(schema);
+        }
```

Comment:
I'll try this, I've had trouble casting DynamoUploadSchema to UploadSchema, which is why I have ended up iterating in cases like this.